homepage/views.py

```diff
@@ -53,3 +56,26 @@ def add_review(request):
     else:
         form = ReviewForm()
     return render(request, 'homepage/add_review.html', {'form': form})
+
+
+def sign_in(request):
+    if request.method == "POST":
+        form = AuthenticationForm(request, data=request.POST)
+
+        if form.is_valid():
+            username = form.cleaned_data.get('username')
+            password = form.cleaned_data.get('password')
+            user = authenticate(username=username, password=password)
+
+            if user is not None:
+                login(request, user)
+                messages.info(request, f"You are now logged in as {username}.")
+                return redirect("landing")
+            else:
+                messages.error(request, "Invalid username or password.")
+
+        else:
+            messages.error(request, "Invalid username or password.")
+
```

Comment:
I see you've used the messages framework from django, but is it displayed anywhere?
If I understand your structure correctly, then there is supposed to be a display of this error message at the marked line in the sign_in html file.

Reply:
Error messages are only partly implemented and will be fully implemented on a separate PR with other features
I've added a comment in the PR description for more clarity.

Reply:
As I've said in the PR #100 - this is implemented enough to display, so please display.

Reply:
@VeraTs Added implementation for messages in the HTML pages